modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/SourceConfigurer.java

```diff
@@ -172,4 +177,28 @@ private int resolverMaxItemsPerPoll(ResolverSet resolverSet, ValueResolvingConte
       return maxItemsPerPoll;
     }
   }
+
+  private ResolverSetResult buildResolverSetResult(Object source, ValueResolvingContext context) throws MuleException {
+    ResolverSetResult resolverSetResult = resolverSet.resolve(context);
+    Optional<Field> field = Arrays.stream(source.getClass().getDeclaredFields())
+        .filter(f -> f.getType().getName().equals(org.mule.runtime.extension.api.tx.SourceTransactionalAction.class.getName()))
+        .findFirst();
+    if (field.isPresent() && resolverSet.getResolvers().get(field.get().getName()) != null) {
+      return overwriteResolverResult(field.get().getName(), resolverSetResult);
```

Comment:
is there a test that covers this overwriting behaviour?

Reply:
Yes, several test in integration tests such as JmsXATXTestCase plus all the tests in the JMS connector.